includes/class-wc-post-data.php

```diff
@@ -326,9 +326,12 @@ public static function delete_post( $id ) {
 				if ( $parent_id = wp_get_post_parent_id( $id ) ) {
 					wc_delete_product_transients( $parent_id );
 				}
+
+				do_action( 'woocommerce_delete_product', $id );
```

Comment:
Thinking out loud; is it possible to hijack the delete event in WP admin and use CRUD instead?